src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,456 @@
+            case TokenTypes.ARRAY_INIT:
+                leftBrace = ast;
+                break;
+            case TokenTypes.INTERFACE_DEF:
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.ANNOTATION_DEF:
+            case TokenTypes.ENUM_DEF:
+            case TokenTypes.ENUM_CONSTANT_DEF:
+            case TokenTypes.LITERAL_NEW:
+                // These tokens are for the fine tuning. For example, you can enable this for
+                // classes, but not for enums.
+                final DetailAST objBlock = ast.findFirstToken(TokenTypes.OBJBLOCK);
+                if (objBlock != null) {
+                    leftBrace = objBlock.getFirstChild();
+                }
+                break;
+            case TokenTypes.OBJBLOCK:
+                // A general case of class/interface/enum/enum field/annotation/anonymous class.
+                leftBrace = ast.getFirstChild();
+                break;
+            case TokenTypes.LITERAL_ELSE:
+                // Special handling for the "else if" case.
+                final DetailAST candidate = ast.getFirstChild();
+                if (candidate.getType() == TokenTypes.SLIST) {
+                    leftBrace = candidate;
+                }
+                break;
+            case TokenTypes.SLIST:
+                // The scope block is a SLIST with another SLIST as the parent.
+                if (ast.getParent().getType() == TokenTypes.SLIST) {
+                    leftBrace = ast;
+                }
+                break;
+            case TokenTypes.LITERAL_SWITCH:
+                // The Switch does not have SLIST unlike the others.
+                leftBrace = ast.findFirstToken(TokenTypes.LCURLY);
+                break;
+            case TokenTypes.CASE_GROUP:
+                // The CASE_GROUP is the parent of 'case', 'default' and SLIST parent.
+                leftBrace = ast.getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            case TokenTypes.LITERAL_CASE:
+            case TokenTypes.LITERAL_DEFAULT:
+                // The 'case' and 'default' are siblings for SLIST parent.
+                leftBrace = ast.getParent().getLastChild().findFirstToken(TokenTypes.SLIST);
+                break;
+            default:
+                // All the rest: for/while/if/STATIC_INIT,lambda,etc.
+                leftBrace = ast.findFirstToken(TokenTypes.SLIST);
+                break;
+        }
+
+        return leftBrace;
+    }
+
+    /**
+     * {@link BlockWrapOption} factory method.
+     *
+     * @param optionStr option name, such as "empty-line-allowed", "empty-line", "no-empty-line"
+     * @return the {@code BlockWrapOption} associated with {@code optionStr}
+     */
+    private static BlockWrapOption parseWrapOption(String optionStr) {
+        try {
+            return BlockWrapOption.valueOf(
+                optionStr.trim().replace('-', '_').toUpperCase(Locale.ENGLISH));
```

Comment:
our other checks use underscores or camel case. I think we should stick to one or the other.

Reply:
They are all inconsistent. The ParenPadCheck has "nospace", the LeftCurlyCheck has "nlow", ReghtCurlyCheck has "alone_or_singleline". It looks odd. I think that all parameter names should follow one style. This is another story, but please share your opinion on the best style of parameter names here. I will do so for this check.